### ISSUE TITLE
Device Width and Fingers in mos generator seems incorrect. 

### ISSUE THREAD

**Issue by christoph-weiser** [RTimothyEdwards/open_pdks#464]
## Problem
The way magic generates mos devices is not respecting the Fingers field. It should divide the total Width by Fingers to give the correct width of each individual finger. 
The total width of the device should still be what was typed in the Width field.

## Implications

The problem is that this bug renders magics SPICE import function unusable without modifying the netlist manually before importing.

## Example

The input to the generator:
![image](https://github.com/user-attachments/assets/1dc9b735-2399-4dd0-96b4-6b42da149e99)

This is the layout generated: 
![image](https://github.com/user-attachments/assets/28e75b40-62c4-4210-b515-b946fdc6c617)

**Comment by christoph-weiser** [comment]
Ahh ok, i assumed the SPICE netlist would be directly plugged into the device generator without any preprocessing. 

That makes sense then, but I think it would be good to indicate the Width means width per finger to avoid confusion, as it differs from what one would typically write in the schematic.

**Comment by RTimothyEdwards** [comment]
Agreed.  Feel free to leave the issue open until I get around to implementing it.
